crates/core/tedge_actors/src/message_boxes.rs

```diff
@@ -209,6 +214,73 @@ impl<Input: Message, Output: Message> SimpleMessageBox<Input, Output> {
     }
 }
 
+#[async_trait]
+impl<Input: Message, Output: Message> ReceiveMessages<Input> for SimpleMessageBox<Input, Output> {
+    async fn try_recv(&mut self) -> Result<Option<Input>, RuntimeRequest> {
+        self.input_receiver.try_recv().await
+    }
+
+    async fn recv_message(&mut self) -> Option<WrappedInput<Input>> {
+        self.input_receiver.recv_message().await
+    }
+
+    async fn recv(&mut self) -> Option<Input> {
+        self.input_receiver.recv().await.map(|message| {
+            self.log_input(&message);
```

Comment:
Okay for now - as this keeps the current behavior.

However, as a follow up PR, we will need to log whatever the recv method. The simpler will be to use a `LoggingReceiver` which wraps a `CombinedReceiver`, implements `ReceiveMessages` and logs received messages.